crates/short-header-proof-provider/src/native.rs

```diff
@@ -69,8 +69,15 @@ impl<Da: DaSpec> ShortHeaderProofProvider for NativeShortHeaderProofProviderServ
                             occ.get_mut().push(block_hash);
                         }
                         std::collections::hash_map::Entry::Vacant(vac) => {
+                            if queried_hashes_map.capacity() == queried_hashes_map.len() {
+                                queried_hashes_map.try_reserve(1).map_err(|e| {
```

Comment:
I checked hashbrown. `HashMap::insert` always reserves +1 elem, `VacantEntry::insert` reserves only when capacity is not enough. So calling try_reserve(1) before `vac.insert` should solve OOM problem.